### DISCUSSION TITLE
Batch Appending and Size Decrease

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7870] [Help]
### ImageMagick version

7.1.1-40

### Operating system, version and so on

Windows 11

### Description

I have 9 1920x1080 images. I want to append 3 of them at a time vertically while decreasing either the output images or each input images by dividing them by 3 (e.g the 3 output images would be 1920x3240, but they would be shrunken down to 640x1080, or the input images would be shrunken down to 640x360). How do I do that? Is there a script for it?

**Comment by fmw42** [comment]
`magick montage image1.suffix ... image9.suffix -tile 3x3 -geometry 640x360+0+0 result.suffix
`

If all the names start the same, then 

`magick montage image*.suffix -tile 3x3 -geometry 640x360+0+0 result.suffix`

**Comment by GeeMack** [comment]
At its very simplest, if the 9 images are the only ones in the directory and can be called into the command with a wildcard, a command like this can resize them and create 3 output images, each with 3 of the original images resized to 1/3 and stacked vertically.

    magick *.png -resize 33.333% -append -crop 1x3@ +repage result-%02d.png

That reads in the 9 images and produces 3 outputs, the first with images 1, 2, and 3 appended vertically, another with images 4, 5, and 6 appended, and the last image contains the original 7, 8, and 9 vertically appended.

**Reply by Jay123210599** [reply]
One question, why use repage?

**Reply by fmw42** [reply]
After -crop, -trim, -rotate etc, you should use +repage to remove the virtual canvas information that is included with PNG, GIF, TIFF output. Otherwise, subsequent processing might give an image the size of the input or offset from the crop or trim, which you might not want. It is not saved by JPEG and most other formats.

**Reply by Jay123210599** [reply]
What if I want my output images to be JPGs?

**Reply by fmw42** [reply]
As I said above, JPEG does not store this information, so +repage can be dropped off your command. But it does no harm to include if later you want to use the command for PNG, GIF or TIFF

**Reply by Jay123210599** [reply]
What if I have 44 images and want to output 22 640x720 images?

**Reply by fmw42** [reply]
For which command, mine or GeeMack?

Also what are your file names?  Do they all have the same prefix followed by numbers and the same suffix? Or all they all random name?  How many tiles for the output? Is it two inputs for each output? That is are you appending two images to make one output so that 44 inputs become 22 outputs?

**Reply by Jay123210599** [reply]
GeeMack's command, they're all png files and have different names. I want stack two images vertically.

**Reply by fmw42** [reply]
GeeMack's command using the wild card * for input will list images alphabetically by name. If that is not how you want to put every two images together, then you will have to list the files or provide a text file with the desired list as input. His command already stacks vertically.

Wait for GeeMack to respond for the needed changes.

**Reply by fmw42** [reply]
You want your images resized to 720x360 so resize them by 1/3 or 33.333%. You start with 44 images and want 22 output. So the command resizes then appends them all vertically, then the crop makes 22 images by pairs from the 44 images.

So try

`magick *.png -resize 33.333% -append -crop 1x22@ +repage result-%02d.png`

GeeMack please correct me if I am wrong.
